pkg/apis/duck/v1/delivery_types.go

```diff
@@ -37,6 +37,13 @@ type DeliverySpec struct {
 	// +optional
 	Retry *int32 `json:"retry,omitempty"`
 
+	// Timeout is the timeout of each single request.
+	// More information on Duration format:
+	//  - https://www.iso.org/iso-8601-date-and-time-format.html
+	//  - https://en.wikipedia.org/wiki/ISO_8601
+	//
+	Timeout *string `json:"timeout,omitempty"`
```

Comment:
No new things until we get to 1.0.

If it is not in the spec, it does not land in core.

https://github.com/knative/specs/search?q=timeout

/hold

Reply:
> No new things until we get to 1.0.

This is IMO an important feature, where this was discussed?

Reply:
This is all @vaikas and I have been talking about for several weeks now. We talked about the bar for new features to be VERY high in the working group updates for the last couple times.

You have to change the spec to change core features, this has been the way since we moved to v1 in api shape (really since v1beta1)/

Reply:
Then I'll go ahead and add the timeout to the spec

Reply:
Spec PR: https://github.com/knative/specs/pull/15

Reply:
This comment doesn't apply anymore per experimental feature process